src/charm.py

```diff
@@ -387,18 +412,41 @@ def get_mongos_host(self) -> str:
         """
         return self.unit_host(self.unit)
 
-    def get_mongos_hosts(self) -> Set:
-        """Returns the host for mongos as a str.
+    def get_ext_mongos_hosts(self) -> Set:
+        """Returns the K8s hosts for mongos.
 
-        The host for mongos can be either the Unix Domain Socket or an IP address depending on how
-        the client wishes to connect to mongos (inside Juju or outside).
+        Note: for external connections it is not enough to know the external ip, but also the
+        port that is associated with the client.
         """
-        hosts = {self.unit_host(self.unit)}
-        for unit in self.peers_units:
+        hosts = set()
+        for unit in self.get_units():
+            unit_ip = self.node_port_manager.get_node_ip(unit.name)
```

Comment:
We are not checking on `unit_ip is None` given it is the case on `ApiError.status_code == 403`, which will effectively add a host of the form `None:port`

We are also not handling exceptions of `get_node_id / get_node_port`. Is this intended?

Reply:
> We are not checking on unit_ip is None given it is the case on ApiError.status_code == 403, which will effectively add a host of the form None:port

Good points Mehdi. This function should certainly be updated. We must make changes on VM side [PR here](https://github.com/canonical/mongodb-operator/pull/483)

Reply:
I think the linked PR is not the right one on the topic

Reply:
It is - the changes you requested require that we raise and error here and then handle it in the provider side?